websocket-compiler-plugin/src/main/java/io/ballerina/stdlib/websocket/plugin/WebSocketUpgradeServiceValidatorTask.java

```diff
@@ -74,6 +84,43 @@ private boolean isListenerBelongsToWebSocketModule(TypeSymbol listenerType) {
         return false;
     }
 
+    private void validateListenerArguments(SyntaxNodeAnalysisContext context,
+            ListenerInitExpressionNodeVisitor visitor) {
+        visitor.getExplicitNewExpressionNodes()
+                .forEach(explicitNewExpressionNode -> {
+                    SeparatedNodeList<FunctionArgumentNode> functionArgs = explicitNewExpressionNode
+                            .parenthesizedArgList().arguments();
+                    verifyListenerArgType(context, explicitNewExpressionNode.location(), functionArgs);
+                });
+
+        visitor.getImplicitNewExpressionNodes()
+                .forEach(implicitNewExpressionNode -> {
+                    Optional<ParenthesizedArgList> argListOpt = implicitNewExpressionNode.parenthesizedArgList();
+                    if (argListOpt.isPresent()) {
+                        SeparatedNodeList<FunctionArgumentNode> functionArgs = argListOpt.get().arguments();
+                        verifyListenerArgType(context, implicitNewExpressionNode.location(), functionArgs);
+                    }
+                });
+    }
+
+    private void verifyListenerArgType(SyntaxNodeAnalysisContext context, NodeLocation location,
+            SeparatedNodeList<FunctionArgumentNode> functionArgs) {
+        if (functionArgs.size() >= 2) {
+            PositionalArgumentNode firstArg = (PositionalArgumentNode) functionArgs.get(0);
+            PositionalArgumentNode secondArg = (PositionalArgumentNode) functionArgs.get(1);
+            SyntaxKind firstArgSyntaxKind = firstArg.expression().kind();
+            SyntaxKind secondArgSyntaxKind = secondArg.expression().kind();
+            if (!(firstArgSyntaxKind == SyntaxKind.NUMERIC_LITERAL && (
+                    secondArgSyntaxKind == SyntaxKind.SIMPLE_NAME_REFERENCE
+                            || secondArgSyntaxKind == SyntaxKind.MAPPING_CONSTRUCTOR))) {
+                DiagnosticInfo diagnosticInfo = Utils.getDiagnosticInfo(
+                        PluginConstants.CompilationErrors.INVALID_LISTENER_INIT_PARAMS);
+                context.reportDiagnostic(DiagnosticFactory.createDiagnostic(diagnosticInfo, location,
+                        WebSocketConstants.PACKAGE_WEBSOCKET));
```

Comment:
Since this is a common functionality for reporting errors this could be moved into a one utility function. You could follow [this](https://github.com/ballerina-platform/module-ballerina-websub/blob/a6e1d85c4524e3bdf63e7b6680bbc5b300c0b38f/websub-compiler-plugin/src/main/java/io/ballerina/stdlib/websub/task/AnalyserUtils.java#L40) sample.

Reply:
Fixed with https://github.com/ballerina-platform/module-ballerina-websocket/pull/137/commits/59a54eefb2efd0d189768589979326b1ba914d84